src/coins.h

```diff
@@ -295,7 +297,7 @@ struct CCoinsCacheEntry
     CCoinsCacheEntry() : coins(), flags(0) {}
 };
 
-typedef boost::unordered_map<uint256, CCoinsCacheEntry, CCoinsKeyHasher> CCoinsMap;
+typedef boost::unordered_map<uint256, CCoinsCacheEntry, SaltedTxidHasher> CCoinsMap;
```

Comment:
Does std/boost::unordered_map guarantee that this will not create SaltedTxidHashers willy-nilly, using different salts?